File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> charSet = new HashSet<>();

        int maxlen = 0;
        int left = 0;

for(int right = 0; right<s.length(); right++){
while(charSet.contains(s.charAt(right))){
    charSet.remove(s.charAt(left));
    left++;
}

charSet.add(s.charAt(right));
maxlen = Math.max(maxlen, right -left + 1);
}
return maxlen;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<Char> set = new HashSet<Char>();
        int left = 0, maxLen = 0;

        for(int right = 0; right < s.Length; right++){
            while(set.Contains(s[right])){
                set.Remove(s[left]);
                left++;
            }
            set.Add(s[right]);
            maxLen = Math.Max(maxLen, right - left + 1);
        }
        return maxLen;
    }
}